src/showingpreviously/cinemas/glasgow_film_theatre.py

```diff
@@ -0,0 +1,142 @@
+
+
+films_cache = {}
+
+
+class GlasgowFilmTheatre(ChainArchiver):
+    def get_showings(self):
+        global films_cache
+        films_cache = {}
+        showings = []
+        req = get_response(WHATS_ON_URL)
+        soup = bs4.BeautifulSoup(req.text, 'html.parser')
+        for showings_on_date in soup.find_all('div', class_='show-strip-group'):
+            for showing_film in showings_on_date.find_all('div', class_='show-strip'):
+                film_heading = showing_film.find('div', class_='show-strip-heading')
+                film, film_attributes = film_from_heading(film_heading)
+                if 'Audio description available' in showing_film.text:
+                    film_attributes['audio-described'] = True
+                for showing_item in showing_film.find_all('li', class_='instance'):
+                    if 'instance-online' in showing_item['class']:
+                        # ignore online-only showings
+                        continue
+                    showing_datetime = datetime.datetime.fromtimestamp(int(showing_item['data-timestamp']))
+                    showing_screen = Screen(showing_item['data-instance-venue'])
+                    showing_attributes = get_showing_attributes(showing_item)
+                    showing = Showing(
+                        film=film,
+                        time=showing_datetime,
+                        chain=CHAIN,
+                        cinema=CINEMA,
+                        screen=showing_screen,
+                        json_attributes={**film_attributes, **showing_attributes}
+                    )
+                    showings.append(showing)
+        return showings
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def parse_film_title(film_title: str) -> (str, dict[str, any]):
+    # use lowercase
+    film_title = film_title.lower()
+    film_attributes = {'format': []}
+    # strip the rating from the film
+    # note: the rating is always last
+    if film_title.find('(') != -1:
+        film_title = film_title[:film_title.find('(')]
+    # check for ignore keywords
+    for keyword in FILM_TITLE_IGNORES:
+        if keyword.lower() in film_title:
+            film_title = film_title.replace(keyword.lower(), '')
+    # check for 4K screenings
+    if '4k' in film_title:
+        film_attributes['format'] += ['digital', '4K']
+        film_title = film_title.replace('4k', '').strip()
+    # check for 35mm screenings
+    if '35mm' in film_title:
+        film_attributes['format'].append('35mm')
+        film_title = film_title.replace('35mm', '').strip()
+    # if no format is given, assume digital
+    if len(film_attributes['format']) == 0:
+        film_attributes['format'] = ['digital']
+    # clean up the title by removing trailing characters,
```

Comment:
In November the GFT have [Noirvember](https://glasgowfilm.org/shows/noirvember) - it would be good to strip that out of the film name.

Additionally the Noirvember films have ` + introduction` in their title - we could take these out too? Maybe have an attribute for whether the screening is having an introduction?